[TT-9648]: Document Tyk Cloud SSO(single sign-on)

komuw opened the pull request:
### Jira DX PR ticket
- https://tyktech.atlassian.net/browse/TT-9648

### Preview Link
<!-- Add a direct preview link to make it easier for the reviewer to review. Best to add a few direct links to the pages in the PR -->
- https://deploy-preview-3154--tyk-docs.netlify.app/docs/tyk-cloud/teams--users/single-sign-on/

### Description
<!-- 1. Describe your changes in detail. Why is this change required? What problem does it solve?
     2. Please explain the change to the reviewer. Pay special attention to changes that are hard to spot, 
       for example:
       2.1. Name change in the file name or directory name - please flag it out since it’s hard 
            to compare text after such a change 
       2.2. Terminology change - flag it out to make sure the reviewer is aware before approving
     3. @mentions of the person to review the proposed changes. They need to be able to know the topic well in order to approve it.
-->
- Document Tyk Cloud SSO(single sign-on).
In Tyk Cloud, we recently(Ara v1.9.0) released some changes that enables costumers to use SSO using their preferred identity providers.
That feature is not yet documented in https://tyk.io/docs/. This PR does that.
- There was already a section in the Tyk docs for managing [Teams & Users](https://tyk.io/docs/tyk-cloud/teams--users/managing-teams/), this PR creates a new page under that section.

#### Screenshots (if appropriate)

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply -->

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR
<!-- Label your PR according to the type of changes that your code introduces. This ensures that we know how/when to publish the PR. These are the options:
- Fixing typo (please merge to production) - add the label `now`
- Documenting a new feature (please merge to production) - add the label `now`
- Documentation for future release (please do not merge to production) - add label `future release` and label of the release
- [Something else (please add if needs merging to production or not) - add label according to the above logic -->

komuw commented:
Thanks to Mitja, since this documentation is basically taken from https://tyktech.atlassian.net/wiki/spaces/EN/pages/2036564149/Single+sign-on

dcs3spp reviewed (changes requested):
PR looks good to me komu. I have made some minor suggestions and will merge and release once these have been completed and a peer review has been done

komuw commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,77 @@
+
+  {{< img src="/img/cloud/cloud-sso-add-profile-form.png" alt="Tyk Cloud SSO add profile form" >}}
+
+The fields to fill and their explanation are:
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/86ffc05d2871fb79c50bb4490f1e655a5df7704e

komuw commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,77 @@
+  {{< img src="/img/cloud/cloud-sso-add-profile-form.png" alt="Tyk Cloud SSO add profile form" >}}
+
+The fields to fill and their explanation are:
+| Field name             | Explanation                                                                                                                     |
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/e89ec3b6ba464ea2b735e4c7fd96e23be57efa8c

komuw commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,78 @@
+
+
+As illustrated below an authentication and callback URL will be generated, once the new profile has been added. You need to add these URLs to the configuration of your chosen IdP provider.
+The Auth URL is your custom SSO login URL whereas Callback URL is the URL that the SSO provider will callback to, after the user authenticates.
```
@mitjaziv is my explanation for auth & callback url clear & correct?

mitjaziv replied:
There are multiple URL's:
- Auth/Login URL: Is the URL for staring the Login Flow 
- Discover URL: Is the URL for IDP discovery, reading configuration etc...
- Callback URL: Is the URL called after successful authentication
- Redirect URL: Is the URL where user gets redirected after successful authentication/authorization

komuw replied:
i've updated the definitions in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/8987e7436216a3256bcf7307f4bfa43bfeb25552  

I've only included auth url and callback url since they are the only ones that we show to the user on the details page.

komuw commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,77 @@
+| Only registered users  | It is a check-box that defines which users are allowed to use SSO. If checked, only users who are already registered in Tyk Cloud are allowed to login with SSO. Else if un-checked, new authenticated user will be added to Tyk Cloud in Default user group. |
+
+
+Once the new profile has been added, you also need to add the authentication URL & callback URL to the configuration of your chosen IdP provider.
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/c339d7ecab0fe281e54e9f2a08485cf60e0bd8df

SonjaChevre commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,78 @@
+---
+title: "Tyk Cloud Single sign-on(SSO)"
```
Tyk Cloud Single Sign-On (SSO) or Tyk Cloud single sign-on (SSO)

komuw replied:
fixed inhttps://github.com/TykTechnologies/tyk-docs/pull/3154/commits/907b9e1f57b0dfff7a0c15314243aea94ffad450

SonjaChevre commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,78 @@
+## What is SSO?
+Single sign-on (SSO) is an authentication process which allows users to login to multiple services and applications with one set of credentials.
+
+These credentials are stored with an Identity Provider(IdP). An Identity Provider (IdP) is a service that stores and manages digital identities. Companies can use these services to allow their employees or users to connect with the resources they need. 
```
These credentials **are securely stored** with an Identity Provider (IdP).

komuw replied:
fixed in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/907b9e1f57b0dfff7a0c15314243aea94ffad450

SonjaChevre commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,78 @@
+---
+
+## What is SSO?
+Single sign-on (SSO) is an authentication process which allows users to login to multiple services and applications with one set of credentials.
```
Single Sign-On (SSO) is an authentication process that empowers users to access various services and applications using a single set of credentials. This streamlines the login experience by eliminating the need to remember multiple usernames and passwords for different platforms.

komuw replied:
fixed in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/907b9e1f57b0dfff7a0c15314243aea94ffad450

SonjaChevre commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,78 @@
+This functionality is a preview feature and may change in subsequent releases.
+
+To be able to configure Single Sign-On authentication, an SSO entitlement needs to be enabled in the subscription plan. 
+If you are interested in getting access, contact us at [support@tyk.io](<mailto:support@tyk.io?subject=Tyk Cloud Single sign on>)
```
contact your account manager or reach out to our [support]

komuw replied:
fixed in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/767c43c1684b1a2dedc6303f789aac2adac2dee8

SonjaChevre commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,78 @@
+{{< /note >}}
+
+## Add new SSO profile
+To add a new SSO profile, login to Tyk Cloud Dashboard, navigate to the _Profile_ list and click on the _ADD PROFILE_ button.
```
login to Tyk Cloud
(not Tyk Cloud Dashboard, I don't think we name it like this in the public doc)

komuw replied:
fixed in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/439b40c26cc1e592657dfc1ddfb301684600fa4a

SonjaChevre reviewed (commented):
looking really good - I like how detailed it is!
I have added a couple of notes for small improvements.

komuw commented:
> looking really good - I like how detailed it is! I have added a couple of notes for small improvements.

Thanks for reviewing. I've addressed the feedback.

mitjaziv commented on tyk-docs/content/tyk-cloud/teams-&-users/single-sign-on.md:
```diff
@@ -0,0 +1,78 @@
+| Provider name          | Used to distinguish between different SSO providers.                                                                      |
+| Client ID              | Used for client authentication with the IdP provider. The value can be found in your chosen IdP provider's configuration. |
+| Client Secret          | Used for client authentication with the IdP provider. The value can be found in your chosen IdP provider's configuration.     |
+| Discovery URL          | Used to initialise the authentication flow. This URL can be found in your chosen IdP provider's configuration.  |
```
Discovery URL is used for reading IDP's openid configuration.

komuw replied:
i've updated the definitions in https://github.com/TykTechnologies/tyk-docs/pull/3154/commits/8987e7436216a3256bcf7307f4bfa43bfeb25552

mitjaziv reviewed (approved):
Thank you Komu, looks great! 🚀

dcs3spp commented:
/release to release-5.1
